### PR TITLE
Only sleep if there are no tasks

### PR DESCRIPTION
**Pull request by mgax**
I was running the DB worker on a queue full of tasks, and was wondering why it was only processing one task per second. It's bad form to sleep when there is work to do.

### CODE REVIEW

**Comment by RealOrangeOne** [comment]
I'm not sure how I feel about this, as it changes the definition of "interval". It could be surprising to some people. This change increases throughput, but also adds DB load.

It's also arguably a breaking change.

**Comment by mgax** [comment]
Fair points; on the other hand, I don't think it's common/expected for a task queue to throttle its throughput to keep resource use under control. If a user really wants this, they could add a `time.sleep()` themselves, either in the task, or in a post-execute hook.

File: django_tasks/backends/database/management/commands/db_worker.py [hunk 1]
```diff
@@ -91,6 +91,9 @@ def start(self) -> None:
                         # "claim" the task, so it isn't run by another worker process
                         task_result.claim()
 
+                    else:
+                        time.sleep(self.interval)
```
Comment (RealOrangeOne) [inline django_tasks/backends/database/management/commands/db_worker.py:95]: Issue: This holds the locking transaction open longer than necessary, which will affect throughput. This should probably be in the condition below, once the transaction has closed.

Comment (mgax) [reply django_tasks/backends/database/management/commands/db_worker.py]: Whoops, good point! Fixed.

File: django_tasks/backends/database/management/commands/db_worker.py [hunk 2]
```diff
@@ -94,19 +94,16 @@ def start(self) -> None:
                 if task_result is not None:
                     self.run_task(task_result)
 
+                else:
+                    time.sleep(self.interval)
```
Comment (RealOrangeOne) [inline django_tasks/backends/database/management/commands/db_worker.py:98]: I'm not sure moving the `sleep` is the right approach. Moving it here sets `running_task` longer than necessary.

Keeping it after the loop probably makes sense. But perhaps it should be `if self.running and task_result is None`?

Comment (mgax) [reply django_tasks/backends/database/management/commands/db_worker.py]: That makes sense. I've changed the `if` condition, and did a quick test, it's working as expected.

**Comment by RealOrangeOne** [comment]
Change looks good now! Although I'm a little concerned about it being a breaking change. The help text for `--interval` mentions checking for new tasks. I can't think of a better name than "interval", but the text should be updated to match its new use.

**Comment by mgax** [comment]
> Although I'm a little concerned about it being a breaking change.

That's a good argument for the next release being `0.4.0` :)

> but the text should be updated to match its new use.

How about _The interval (in seconds) to wait, when there are no tasks in the queue, before checking for tasks again_?

**Comment by RealOrangeOne** [comment]
> The interval (in seconds) to wait, when there are no tasks in the queue, before checking for tasks again?

Works for me!

> the next release being `0.4.0`

It probably will be. I've got some other bigger changes in the work which warrant a larger release bump.

**Comment by mgax** [comment]
Alright; I've updated the `sleep` condition and the help text for the CLI.

**Pull request merged**